packages/libs/error-stack/tests/test_compatibility.rs

```diff
@@ -7,33 +7,16 @@
 
 mod common;
 
-#[cfg(all(rust_1_65, feature = "std"))]
-use std::backtrace::Backtrace;
-#[cfg(feature = "eyre")]
-use std::sync::Once;
-#[cfg(all(rust_1_65, feature = "std", feature = "eyre"))]
-use std::{backtrace::BacktraceStatus, error::Error, ops::Deref};
-
 use common::*;
 use error_stack::compat::IntoReportCompat;
 
-#[cfg(all(rust_1_65, feature = "std", feature = "eyre"))]
-fn has_backtrace<E: Deref<Target = dyn Error + Send + Sync>>(err: &Result<(), E>) -> bool {
-    err.as_ref()
-        .unwrap_err()
-        .deref()
-        .request_ref::<Backtrace>()
-        .filter(|bt| bt.status() == BacktraceStatus::Captured)
-        .is_some()
-}
-
 #[cfg(all(rust_1_65, feature = "std"))]
-fn remove_backtrace_context(messages: &mut Vec<String>) {
-    // anyhow/eyre has a backtrace, this means we don't add it ourselves,
-    // therefore we need to remove the context (if it supports backtrace)
-    let last = messages.pop().unwrap();
-    messages.pop();
-    messages.push(last)
+fn remove_opaque_frames(messages: &mut Vec<String>) {
+    /// Depending on how the backtrace is generated, it will appear at different locations. To
+    /// simplify the general tests, we remove backtraces, they are tested explicitly in other tests.
+    /// On nightly, anyhow/eyre will capture the backtrace and provide it by `Error::provide`. On
+    /// non-nightly toolchains since 1.65 the backtrace will be captured by `Report.
+    messages.retain(|message| message != "Opaque")
```

Comment:
Flagging/nitpick: This will break if one of the tests uses opaque attachments.

Might it be worth a comment/disclaimer?

Reply:
I added a note in 822d0cf220c559ee279a404dc646dab9b114d504